### ISSUE TITLE
[TalkBack] Duplicate speech during text replacement, e.g. "replaced X with X"

### ISSUE THREAD

**Issue by GoogleCodeExporter** [clc/eyes-free#241]
```
What steps will reproduce the problem?
1.Precondition:talkback on Speak password : On
 WIFi Setting->Input Wifipassword->output "End"

2. [Input field]  read the letters in the right Edit cursor 
Enter the "Hoouse".->Touch the screen to clear the "o" ->Talk back, but the 'o' 
Edit If you move the cursor to the right of the H
If you delete the 'H' will be deleted.

3. Whenever you enter all the input read every syllable and B (a) A (a) replace 
the 'with difficulty hear overlapping.

4.It does not understand Korean, if B (a) A (a), replace "Label is not suitable 
when each input.

What is the expected output? What do you see instead?
Normal operation.

What version of the product are you using? On what operating system?
Android 4.2.1 ,talkback 3.3.*

Please provide any additional information below.

```

Original issue reported on code.google.com by `ezc...@gmail.com` on 23 Apr 2013 at 6:51

**Comment by GoogleCodeExporter** [comment]
```
TalkBack always reads the character to the right of the cursor. Do you mean 
that this does not make sense for Korean text? Please explain what "B (a) A (a) 
replace the 'with difficulty hear overlapping" means in more detail.
```

Original comment by `alanv@google.com` on 24 Apr 2013 at 6:31

**Comment by GoogleCodeExporter** [comment]
```
Korean are formed from combinations of 14 consonants and 10 vowels.
To make a character, '가(GA)', we have to use both a consonants,'ㄱ(G)' and a 
voewl 'ㅏ(A)'. So we push 'ㄱ(G)' and then 'ㅏ(A)' on IME.

Talk Back used to have voice feedback like 'ㄱ(G)' and then '가(GA)' on 
version 3.2.1
However, version 3.3.2 have the voice feedback in different way like '가(GA)' 
replaced '가(GA)'. In this case, it is hard to understand inserted words.
Please fix to have the same spoken feedback of IME with version 3.2.1
```

Original comment by `lge.stel...@gmail.com` on 25 Apr 2013 at 2:07

**Comment by GoogleCodeExporter** [comment]
```
So the issue is that TalkBack speaks "X replaced with X" where both values are 
identical, and the expected result is that TalkBack speaks "X" where the value 
occurs only once?
```

Original comment by `alanv@google.com` on 30 Apr 2013 at 7:50
- Changed title: **[TalkBack] Spoken feedback for text entry from Korean IME is confusing**

**Comment by GoogleCodeExporter** [comment]
```
Yes, our expected result is same behavior with v3.2.1.
```

Original comment by `lge.stel...@gmail.com` on 30 Apr 2013 at 8:09

**Comment by GoogleCodeExporter** [comment]
```
Is there any updates???
```

Original comment by `lge.stel...@gmail.com` on 7 May 2013 at 1:46

**Comment by GoogleCodeExporter** [comment]
Original comment by `alanv@google.com` on 7 May 2013 at 6:31
- Changed title: **[TalkBack] Duplicate speech during text replacement, e.g. "replaced X with X"**
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
Original comment by `caseybur...@google.com` on 29 May 2013 at 8:02

**Comment by GoogleCodeExporter** [comment]
```
Should be fixed in 3.4.0_b1.
```

Original comment by `caseybur...@google.com` on 4 Jun 2013 at 1:35
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Hi

We still reproduce this issue on TalkBack Version 3.5.
Could you double check it?
```

Original comment by `lge.stel...@gmail.com` on 11 Oct 2013 at 1:23

**Comment by GoogleCodeExporter** [comment]
```
Hi, 

it's critical issues in Korea.

We should uninstall the updated Talkback due to this issue so we can't use the 
latest TalkBack features in Korea.
```

Original comment by `jihyoung...@lge.com` on 13 Nov 2013 at 5:04

**Comment by GoogleCodeExporter** [comment]
```
Issue 330 has been merged into this issue.
```

Original comment by `caseybur...@google.com` on 5 Dec 2013 at 6:08
